Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -46,6 +46,8 @@ public class ProductStore: Store {
             retrieveProduct(siteID: siteID, productID: productID, onCompletion: onCompletion)
         case .retrieveProducts(let siteID, let productIDs, let pageNumber, let pageSize, let onCompletion):
             retrieveProducts(siteID: siteID, productIDs: productIDs, pageNumber: pageNumber, pageSize: pageSize, onCompletion: onCompletion)
+        case .retrieveProductsInCacheSortedByDescendingPopularity(let siteID, let onCompletion):
+            retrieveProductsInCacheSortedByDescendingPopularity(siteID: siteID, onCompletion: onCompletion)
```

Comment:
Is this action too specific? Perhaps something like the following would be more re-usable and narrow down its responsibility:

```
enum ProductsInCacheSortOrder {
  case descendingPopularity
  ...
}

retrieveAllProductsInCache(siteID: Int64, sortOrder: ProductsInCacheSortOrder?, onCompletion: ([Product]) -> Void)
```

I don't have any specific case for re-usability at the moment, but I'll comment further in the function implementation.

Reply:
Thanks for your comment, that's a good point. As you mentioned, we do not have any specific case for re-usability, neither now nor in the coming future. Making it generic would make the implementation more complex and thus more error-prone, without adding any clear benefit.